dask_ml/model_selection/_hyperband.py

```diff
@@ -24,59 +26,73 @@ def _get_nr(R, eta=3):
     return list(map(int, N)), list(map(int, R)), brackets
 
 
-def _partial_fit(model_and_meta, X, y, meta=None, fit_params={},
+def _partial_fit(model_meta_future, X, y, meta=None, fit_params={},
                  dry_run=False, **kwargs):
+    start = time()
     assert isinstance(X, np.ndarray)
     assert y is None or isinstance(y, np.ndarray)
-    model, m = model_and_meta
+    model, m = model_meta_future
     if meta is None:
         meta = m
     while meta['iterations'] < meta['partial_fit_calls']:
         if not dry_run:
             model.partial_fit(X, y, **fit_params)
         meta["iterations"] += 1
+    meta['mean_fit_time'] += time() - start
     return model, meta
```

Comment:
We should ensure that this function (and all others) do not mutate their inputs in-place.  Ideally this would clone the model, then do a `partial_fit` on that cloned model and return it, leaving the input model untouched in case someone else has to use it.

Reply:
I'm using `deepcopy` on the models (sklearn's `clone` won't work: it returns an unfitted instance).

Copying models may be expensive, so I put in metrics to view how much time is spent copying and a parameter to control whether to copy or not.

Reply:
Understood.  This is a common-ish cost of distributed computing.  Also, you may want to familiarize yourself with the profiler in dask.distributed.  It may already be sufficient to tell how expensive this is.